local setup fails due to FelixConfiguration timeout

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
-->
/area dev-productivity
/kind enhancement

**What would you like to be added**:
add condition check `daemonset` `kube-system\calico-node` before executing `FelixConfiguration`. 
**Why is this needed**:
In local development `make kind up`, It seems `FelixConfiguration` timeout is due to `kube-system\calico-node` not being ready sometimes.  even change `max_retries=120` to higher. Sometimes is requested around 7~8 mins become to running status for `kube-system\calico-node`
```
make kind-up
...

Waiting for FelixConfiguration to be created...
Error: FelixConfiguration 'default' not found or patch failed after 200 attempts.
make: *** [kind-up] Error 1
```
```
k get felixconfigurations.crd.projectcalico.org  -A
No resources found

ks get pod
NAME                                                   READY   STATUS              RESTARTS   AGE
calico-kube-controllers-55fc758c88-lhzb8               0/1     ContainerCreating   0          7m35s
calico-node-82qmf                                      0/1     Init:2/3            0          7m35s
coredns-6d4b75cb6d-4t82p                               0/1     ContainerCreating   0          8m54s
coredns-6d4b75cb6d-nwv78                               0/1     ContainerCreating   0          8m54s
etcd-gardener-local-control-plane                      1/1     Running             0          9m14s
kube-apiserver-gardener-local-control-plane            1/1     Running             0          9m12s
kube-controller-manager-gardener-local-control-plane   1/1     Running             0          9m12s
kube-proxy-2pjvn                                       1/1     Running             0          8m54s
kube-scheduler-gardener-local-control-plane            1/1     Running             0          9m19s
metrics-server-7b975bf89b-jsr9r                        0/1     ContainerCreating   0          7m28s
```

--- a/hack/kind-up.sh
+++ b/hack/kind-up.sh
@@ -128,10 +128,10 @@ kubectl get nodes -l node-role.kubernetes.io/control-plane -o name |\
 
 # Allow multiple shoot worker nodes with calico as shoot CNI: As we run overlay in overlay ip-in-ip needs to be allowed in the workload.
 # Unfortunately, the felix configuration is created on the fly by calico. Hence, we need to poll until kubectl wait for new resources
-# (https://github.com/kubernetes/kubernetes/issues/83242) is fixed. (2 minutes should be enough for the felix configuration to be created.)
+# (https://github.com/kubernetes/kubernetes/issues/83242) is fixed. (10 minutes should be enough for the felix configuration to be created.)
 echo "Waiting for FelixConfiguration to be created..."
 felix_config_found=0
-max_retries=120
+max_retries=600
 for ((i = 0; i < max_retries; i++)); do
   if kubectl get felixconfiguration default > /dev/null 2>&1; then
     if kubectl patch felixconfiguration default --type merge --patch '{"spec":{"allowIPIPPacketsFromWorkloads":true}}' > /dev/null 2>&1; then

